examples/LaravelSampleApp/app/Http/Controllers/GoogleAdsApiController.php

```diff
@@ -0,0 +1,199 @@
+     *
+     * @param Request $request the HTTP request
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @return View the view to redirect to after processing
+     */
+    public function showReportAction(
+        Request $request,
+        GoogleAdsClient $googleAdsClient
+    ): View {
+        if ($request->method() === 'POST') {
+            // Retrieves the form inputs.
+            $clientCustomerId = $request->input('clientCustomerId');
+            $reportType = $request->input('reportType');
+            $reportRange = $request->input('reportRange');
+            $entriesPerPage = $request->input('entriesPerPage');
+
+            // Retrieves the list of metric fields to select filtering out the static ones.
+            $selectedFields = array_values(
+                $request->except(
+                    [
+                        '_token',
+                        'clientCustomerId',
+                        'reportType',
+                        'reportRange',
+                        'entriesPerPage'
+                    ]
+                )
+            );
+
+            // Merges the list of metric fields to the resource ones that are selected by default.
+            $selectedFields = array_merge(
+                self::$REPORT_TYPE_TO_DEFAULT_SELECTED_FIELDS[$reportType],
+                $selectedFields
+            );
+
+            // Builds the GAQL query.
+            $query = sprintf(
+                "SELECT %s FROM %s WHERE metrics.impressions > 0 AND segments.date " .
+                "DURING %s LIMIT 100",
+                join(", ", $selectedFields),
+                $reportType,
+                $reportRange
+            );
+
+            // Searches the results using streaming.
+            /** @var GoogleAdsServerStreamDecorator $stream */
+            $stream = $googleAdsClient->getGoogleAdsServiceClient()->searchStream(
+                $clientCustomerId,
+                $query
+            );
+
+            // Fetches all the results.
+            $results = [];
+            foreach ($stream->readAll() as $response) {
+                /** @var SearchGoogleAdsStreamResponse $response */
+                foreach ($response->getResults() as $googleAdsRow) {
+                    /** @var GoogleAdsRow $googleAdsRow */
+                    // Converts each result as a POPO using JSON.
+                    $results[] = json_decode($googleAdsRow->serializeToJsonString(), true);
+                }
+            }
+            $collection = collect($results);
+
+            // Updates the session with all the information that is necessary to process any
+            // following GET requests (report result pages).
+            $request->session()->put('selectedFields', $selectedFields);
+            $request->session()->put('entriesPerPage', $entriesPerPage);
+            $request->session()->put('collection', collect($collection));
```

Comment:
Why do we need two collect here?
collect(collect($results))?

Reply:
Good catch!